include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -47,6 +51,28 @@ class SubAllocatedDescriptorSet : public core::IReferenceCounted
 			m_composed->m_totalDeferredFrees -= getWorstCaseCount();
 		}
 
+		DeferredFreeFunctor(const DeferredFreeFunctor& other) = delete;
+		DeferredFreeFunctor& operator=(const DeferredFreeFunctor& other) = delete;
+		inline DeferredFreeFunctor& operator=(DeferredFreeFunctor&& other)
+		{
+			m_composed = other.m_composed;
+			m_addresses = other.m_addresses;
+			m_binding = other.m_binding;
+			return *this;
+		}
+
+		// This is needed for the destructor of TimelineEventHandlerST
+		// Don't call this directly
+		// TODO: Find a workaround for this
+		inline void operator()()
+		{
+			core::vector<IGPUDescriptorSet::SDropDescriptorSet> nulls(m_addresses->size());
+			auto ptr = nulls.data();
+			operator()(ptr);
+			auto size = ptr - nulls.data();
+			m_composed->m_logicalDevice->nullifyDescriptors({nulls.data(),size_type(size)});
```

Comment:
let it be, but assert(false) at the top, then you'll know you were supposed to drain the timeline first